library/core/src/panic/panic_info.rs

```diff
@@ -161,13 +161,8 @@ impl fmt::Display for PanicInfo<'_> {
     fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
         formatter.write_str("panicked at ")?;
         self.location.fmt(formatter)?;
-        if let Some(message) = self.message {
-            formatter.write_str(":\n")?;
-            formatter.write_fmt(*message)?;
-        } else if let Some(payload) = self.payload.downcast_ref::<&'static str>() {
-            formatter.write_str(":\n")?;
-            formatter.write_str(payload)?;
-        }
+        formatter.write_str(":\n")?;
+        formatter.write_fmt(self.message)?;
         // NOTE: we cannot use downcast_ref::<String>() here
```

Comment:
This comment can be removed, now that this function no longer does any downcasting.